cmake/modules/FindNetCDF.cmake

```diff
@@ -2,135 +2,115 @@
+    file(STRINGS "${NETCDF_INCLUDE_DIRS}/netcdf_meta.h" _netcdf_version_lines
+      REGEX "#define[ \t]+NC_VERSION_(MAJOR|MINOR|PATCH|NOTE)")
+    string(REGEX REPLACE ".*NC_VERSION_MAJOR *\([0-9]*\).*" "\\1" _netcdf_version_major "${_netcdf_version_lines}")
+    string(REGEX REPLACE ".*NC_VERSION_MINOR *\([0-9]*\).*" "\\1" _netcdf_version_minor "${_netcdf_version_lines}")
+    string(REGEX REPLACE ".*NC_VERSION_PATCH *\([0-9]*\).*" "\\1" _netcdf_version_patch "${_netcdf_version_lines}")
+    string(REGEX REPLACE ".*NC_VERSION_NOTE *\"\([^\"]*\)\".*" "\\1" _netcdf_version_note "${_netcdf_version_lines}")
+    set(NETCDF_VERSION "${_netcdf_version_major}.${_netcdf_version_minor}.${_netcdf_version_patch}${_netcdf_version_note}")
+    unset(_netcdf_version_major)
+    unset(_netcdf_version_minor)
+    unset(_netcdf_version_patch)
+    unset(_netcdf_version_note)
+    unset(_netcdf_version_lines)
   endif ()
-endif ()
-
-find_path(NETCDF_INCLUDE_DIR
-  NAMES netcdf.h
-  DOC "netcdf include directories"
-  PATHS "${NETCDF_DIR}/include")
-mark_as_advanced(NETCDF_INCLUDE_DIR)
 
-find_library(NETCDF_LIBRARY
-  NAMES netcdf
-  DOC "netcdf library"
-  PATHS "${NETCDF_DIR}/lib"
-  HINTS "${NETCDF_INCLUDE_DIR}/../lib")
-mark_as_advanced(NETCDF_LIBRARY)
-
-if (NETCDF_INCLUDE_DIR)
-  file(STRINGS "${NETCDF_INCLUDE_DIR}/netcdf_meta.h" _netcdf_version_lines
-    REGEX "#define[ \t]+NC_VERSION_(MAJOR|MINOR|PATCH|NOTE)")
-  string(REGEX REPLACE ".*NC_VERSION_MAJOR *\([0-9]*\).*" "\\1" _netcdf_version_major "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_MINOR *\([0-9]*\).*" "\\1" _netcdf_version_minor "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_PATCH *\([0-9]*\).*" "\\1" _netcdf_version_patch "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_NOTE *\"\([^\"]*\)\".*" "\\1" _netcdf_version_note "${_netcdf_version_lines}")
-  set(NetCDF_VERSION "${_netcdf_version_major}.${_netcdf_version_minor}.${_netcdf_version_patch}${_netcdf_version_note}")
-  unset(_netcdf_version_major)
-  unset(_netcdf_version_minor)
-  unset(_netcdf_version_patch)
-  unset(_netcdf_version_note)
-  unset(_netcdf_version_lines)
-
-  FindNetCDF_get_is_parallel_aware("${NETCDF_INCLUDE_DIR}")
+  find_library(NETCDF_LIBRARIES
+    NAMES netcdf
+    DOC "netcdf library"
+    PATHS "${NETCDF_DIR}/lib"
+    HINTS "${NETCDF_INCLUDE_DIRS}/../lib")
+  mark_as_advanced(NETCDF_LIBRARIES)
 endif ()
 
+#
+# Check if NetCDF is parallel-aware by testing if ${include_dir}/netcdf_par.h exists.
+#
+find_file(NETCDF_PAR_H netcdf_par.h PATHS ${NETCDF_INCLUDE_DIRS})
+if (NETCDF_PAR_H)
+  set(NETCDF_HAS_PARALLEL TRUE)
+else()
+  set(NETCDF_HAS_PARALLEL FALSE)
+  message(WARNING "NetCDF parallel header file ${include_dir}/netcdf_par.h not found.")
+endif()
+
+# Run `cmake ... --log-level=DEBUG` to display these debugging information.
+message(STATUS "NETCDF search stragegy: via ${SEARCH_STRATEGY}")
+message(STATUS "NETCDF_DIR=${NETCDF_DIR}")
```

Comment:
Note that the log levels here are supposed to be `DEBUG` instead of `STATUS`. This was done to easily inspect the build logs in CI. I'll change this back to `DEBUG` once all the requested fixes in this PR have been resolved.